src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java

```diff
@@ -172,23 +170,20 @@ public EntitlementCertificate generateEntitlementCert(Entitlement entitlement, P
     public X509Certificate createX509Certificate(Consumer consumer, Owner owner, Pool pool,
         Entitlement ent, Product product, Set<Product> products,
         List<org.candlepin.model.dto.Product> productModels, BigInteger serialNumber,
-        KeyPair keyPair, boolean useContentPrefix)
+        KeyPair keyPair, ContentPaths contentPaths, PromotedContent promotedContent)
```

Comment:
@Januson @Ceiu Given that ContentPaths and PromotedContent are almost universally passed into methods together, and conceptually, paths are one field of the promoted content, would it not be a good idea to merge the ContentPaths into PromotedContent, and only pass a PromotedContent object around? Something like this:
```
class PromotedContent {
    private final ContentPrefix prefix;
    private final Map<String, EnvironmentContent> contents;
    public String getPath(ProductContent pc) {...};
     // + add all the methods that exist in the current PromotedContent class?
}
```

Reply:
I think I would prefer this overall; but I'm open to other options as well.

Reply:
Done.